FerramAerospaceResearch.Base/UnityJobs/OcclusionJobs.cs

```diff
@@ -0,0 +1,381 @@
+        // Project each of the 8 corners of the bounding box onto the plane.
+        // https://stackoverflow.com/questions/9605556/how-to-project-a-point-onto-a-plane-in-3d/41897378#41897378
+        public void Execute(int index)
+        {
+            float3 normal = math.mul(quaternions[index], new float3(0, 0, 1));
+            float3 origin = float3.zero + math.length(extents) * normal;
+            float2 min = new float2(float.PositiveInfinity, float.PositiveInfinity);
+            float2 max = new float2(float.NegativeInfinity, float.NegativeInfinity);
+            float3 ext = extents / 2;
+            float3 center = boundsCenter;
+            unsafe
+            {
+                float3* arr = stackalloc[]
+                {
+                    center - ext,
+                    new float3(center.x - ext.x, center.y - ext.y, center.z + ext.z),
+                    new float3(center.x - ext.x, center.y + ext.y, center.z - ext.z),
+                    new float3(center.x - ext.x, center.y + ext.y, center.z + ext.z),
+                    new float3(center.x + ext.x, center.y - ext.y, center.z - ext.z),
+                    new float3(center.x + ext.x, center.y - ext.y, center.z + ext.z),
+                    new float3(center.x + ext.x, center.y + ext.y, center.z - ext.z),
+                    center + ext,
+                };
+
+                for (int i = 0; i < 8; i++)
+                {
+                    // Project the bounding box onto the plane
+                    float dist = math.dot(arr[i] - origin, normal);
+                    float3 projection = arr[i] - dist * normal;
+                    // Rotate the projection back towards the vessel alignment
+                    projection = math.mul(math.inverse(quaternions[index]), projection);
+
+                    // The x,y coords of projection are again aligned to the vessel axes.
+                    min.x = math.min(min.x, projection.x);
+                    min.y = math.min(min.y, projection.y);
+                    max.x = math.max(max.x, projection.x);
+                    max.y = math.max(max.y, projection.y);
+                }
+            }
+
+            bounds[index] = new float2(math.abs(max.x - min.x), math.abs(max.y - min.y));
+        }
+    }
+
+    [BurstCompile]
+    public struct SetIntervalAndDimensionsJob : IJobParallelFor
+    {
+        [ReadOnly] public NativeArray<float2> bounds;
+        [ReadOnly] public int maxDim;
+        [ReadOnly] public float resolution;
+        [WriteOnly] public NativeArray<float2> interval;
+        [WriteOnly] public NativeArray<int2> dims;
+
+        public void Execute(int index)
+        {
+            int minXsize = (int)math.ceil(bounds[index].x / resolution);
+            int minYsize = (int)math.ceil(bounds[index].y / resolution);
+            dims[index] = new int2(math.min(maxDim, minXsize), math.min(maxDim, minYsize));
```

Comment:
`new int2(math.ceil(bounds[index] / resolution))` and `math.min(maxDim, minSize)`

Reply:
Implemented in 53809f6